synapse/http/federation/matrix_federation_agent.py

```diff
@@ -236,14 +266,28 @@ class MatrixHostnameEndpoint:
     def __init__(
         self,
         reactor: IReactorCore,
+        proxy_reactor: IReactorCore,
         tls_client_options_factory: Optional[FederationPolicyForHTTPS],
         srv_resolver: SrvResolver,
         parsed_uri: URI,
     ):
         self._reactor = reactor
-
         self._parsed_uri = parsed_uri
 
+        # http_proxy is not needed because federation is always over TLS
+        proxies = getproxies_environment()
+        https_proxy = proxies["https"].encode() if "https" in proxies else None
+        self.no_proxy = proxies["no"] if "no" in proxies else None
+
+        (
+            self.https_proxy_endpoint,
+            self.https_proxy_creds,
+        ) = proxyagent.http_proxy_endpoint(
+            https_proxy,
+            proxy_reactor,
+            tls_client_options_factory or BrowserLikePolicyForHTTPS(),
```

Comment:
I am not sure if `BrowserLikePolicyForHTTPS` is the best default policy.

Reply:
Im not sure what is the best error to raise. `ValueError`, `ConfigError`, `RuntimeError` or anything else?